watertap/unit_models/zero_order/tests/test_nanofiltration_zo.py

```diff
@@ -0,0 +1,162 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Tests for zero-order nanofiltration model
+"""
+import pytest
+
+from pyomo.environ import (
+    ConcreteModel, Constraint, SolverStatus, TerminationCondition, value, Var)
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.testing import initialization_tester
+
+from watertap.unit_models.zero_order import NanofiltrationZO
+from watertap.core.wt_database import Database
+from watertap.core.zero_order_properties import WaterParameterBlock
+
+solver = get_solver()
+
+
+class TestNFZO:
+    @pytest.fixture(scope="class")
+    def model(self):
+        m = ConcreteModel()
+        m.db = Database()
+
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+        m.fs.params = WaterParameterBlock(
+            default={"solute_list": ["sulfur", "toc", "tss"]})
+
+        m.fs.unit = NanofiltrationZO(default={
+            "property_package": m.fs.params,
+            "database": m.db})
+
+        m.fs.unit.inlet.flow_vol.fix(10)
+        m.fs.unit.inlet.conc_mass_comp[0, "sulfur"].fix(1)
+        m.fs.unit.inlet.conc_mass_comp[0, "toc"].fix(2)
+        m.fs.unit.inlet.conc_mass_comp[0, "tss"].fix(3)
+        m.fs.unit.inlet.pressure.fix(1e5)
+        m.fs.unit.inlet.temperature.fix(300)
+
+        return m
+
+    @pytest.mark.unit
+    def test_build(self, model):
+        assert model.fs.unit.config.database == model.db
+
+        assert isinstance(model.fs.unit.electricity, Var)
+        assert isinstance(model.fs.unit.electricity_intensity, Var)
+        assert isinstance(model.fs.unit.deltaP_treated, Var)
+        assert isinstance(model.fs.unit.deltaP_byproduct, Var)
+
+        assert isinstance(model.fs.unit.electricity_consumption, Constraint)
+
+    @pytest.mark.component
+    def test_load_parameters(self, model):
+        data = model.db.get_unit_operation_parameters("nanofiltration")
+
+        model.fs.unit.load_parameters_from_database()
```

Comment:
This is more of a minor suggestion for a future PR, but I think it would be useful to have a method like `list_parameters_from_database` to be able to see what is going to be "loaded" and fixed within the unit model (even though one could probably just see what is inside `data` manually).

Reply:
This is a little trick, because as noted above we are not down-selecting the parameters we retrieve from the database (in the current API at least). I.e. when we do `load_parameters_from_database` we get removals for all components the database supports (we leave it to the specfic instance of the unit model to work out which parameters it actually cares about). So if we display the raw parameter data we would see lots of things we wouldn't actually use.

Reply:
The problem with that is that each unit model would be different, so the unit model developer would need to create that method. Also, you have the issue of keeping things up to date - the values taken from the database may not be the values currently in the model as something might have changed them since. I think it would be far more valuable to show the current values of the parameters, rather than the values loaded from the database.

So, the next question is how to do that?

1. `model.display()` would show all the necessary information (and already exists), but it shows everything so the important information might get obscured.
2. We could add a method to the model which will display a redefined set of variables (basically the IDAES `model.report(0` method), but that would need additional effort from the model developer and would be limited to the pre-defined variables. If a user needed to specify other parameters for their case (e.g. they fixed an outlet concentration rather than a removal fraction), that might not show up. 

So maybe the answer is a combination of the above - the `report()` method for common cases, and a fall back to `display()` if you need more information.

Reply:
My main thought was that upon calling `load_parameters_from_database`, the user should be able to immediately see what was fixed at that moment. If they change fixed values afterwards, that'd be up to them to check via `display()` or a `report()` method (which we'd have to add in). 

However, I think it would be good to add one more optional argument to `load_parameters_from_database` that passes to `set_param_from_data`. The optional arg would be boolean and would either (1) print out which parameters were fixed and to what values/units if true or (2) display nothing if false.

Reply:
@adam-a-a @andrewlee94 - Just a thought on this, there could be an optional boolean argument that if true prints out the variable and its fixed value whenever it is fixed within `load_parameters_from_database`. I think this implementation would be easy, wherever the variable would be fixed it could use a function that has that boolean argument (like verbose or something) that prints the variable, its value, and fixes the variable.

Reply:
@TimBartholomew I agree- that was my thought as well. This way, users could be more informed about what's changing in the model when they apply `load_parameters_from_database`.

Reply:
I have added functionality for this, but rather than add a new argument I have set it up so that all calls to `set_param_from_data` record the action in the logger at the `info_high` level. This way, users can control whether they see this information by setting the logger level.